### PR TITLE
Pass len to eatwhitespaces and eatnewlines

### PR DESCRIPTION
**Pull request by davidanthoff**
Makes a big performance difference

### CODE REVIEW

**Comment by KristofferC** [comment]
I would probably just remove all those default arguments that compute `lastindex` to make sure they aren't hit by accident. If one really wants to it can just be passed explicitly.

**Comment by davidanthoff** [comment]
I agree, that would be much safer!

File: src/csv.jl [hunk 1]
```diff
@@ -533,7 +533,7 @@ function parsefill!(str::AbstractString, opts, rec::RecN{N}, nrecs, cols, colspo
             pos = value(res)
         end
 
-        pos, lines = eatnewlines(str, pos)
+        pos, lines = eatnewlines(str, pos, l)
```
Comment (shashi) [inline src/csv.jl]: What it there are multiple newlines?

**Comment by shashi** [comment]
Oh I thought it's `1`, my bad.

**Pull request merged**